sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,155 @@
+import warnings
+from typing import NamedTuple
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+from sunpy.util.exceptions import SunpyUserWarning
+from sunpy.sun.models import differential_rotation
+
+from sunkit_image.coalignment.decorators import registered_methods
+
+__all__ = ["coalign", "AffineParams"]
+
+
+class AffineParams(NamedTuple):
+    """
+    A 2-element tuple containing scale values defining the image scaling along the x and y axes.
+    """
+    scale: tuple[float, float]
+    """
+    A 2x2 matrix defining the rotation transformation of the image.
+    """
+    rotation_matrix: tuple[tuple[float, float], tuple[float, float]]
+    """
+    A 2-element tuple stores the translation of the image along the x and y axes.
+    """
+    translation: tuple[float, float]
+
+
+def update_fits_wcs_metadata(reference_map, target_map, AffineParams):
```

Comment:
```suggestion
def update_fits_wcs_metadata(reference_map, target_map, affine_params):
```

This is an instance so it should be camel case